src/services/archive.ts

```diff
@@ -0,0 +1,236 @@
+            const academicYear = await this.academicYear(academicYearId)
+
+            let classes = await this.schoolClassArchiveRepository.save(archiveClasses)
+            classes.forEach((newClass, index) => {
+                classes[index].academicYear = academicYear
+            })
+
+            this.schoolClassArchiveRepository.save(classes)
+        }
+
+    }
+
+    public async moveTeacher(academicYearId: number) {
+
+        const archiveTeachers = await this.teacherRepository.find()
+
+        if (archiveTeachers.length <= 0) {
+            throw new Error("No teacher Found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let teachers = await this.teacherArchiveRepository.save(archiveTeachers)
+
+            teachers.forEach((teacher, index) => {
+                teachers[index].academicYear = academicYear
+            })
+
+            this.teacherArchiveRepository.save(teachers)
+        }
+
+    }
+
+    public async moveSubject(academicYearId: number) {
+
+        const archiveSubjects = await this.subjectRepository.find({ relations: ['teacher'] })
+
+        if (archiveSubjects.length <= 0) {
+            throw new Error("No Subjects Found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let subjects = await this.subjectArchiveRepository.save(archiveSubjects)
+
+            subjects.forEach((subject, index) => {
+                subjects[index].academicYear = academicYear
+            })
+
+            this.subjectArchiveRepository.save(subjects)
+        }
+    }
+
+    public async moveExaminations(academicYearId: number) {
+
+        const archiveExams = await this.examinationsRepository.find({ relations: ['teacher'] })
+
+        if (archiveExams.length <= 0) {
+            throw new Error("No Exams Found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let exams = await this.examinationArchiveRepository.save(archiveExams)
+
+            exams.forEach((exam, index) => {
+                exams[index].academicYear = academicYear
+            })
+
+            this.examinationArchiveRepository.save(exams)
+        }
+    }
+
+    public async moveResults(academicYearId: number) {
+
+        const archiveResults = await this.resultRepository.find({ relations: ['exam', 'student'] })
+
+        if (archiveResults.length <= 0) {
+            throw new Error("No results found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let results = await this.resultArchiveRepository.save(archiveResults)
+
+            results.forEach((result, index) => {
+                results[index].academicYear = academicYear
+            })
+
+            this.resultArchiveRepository.save(results)
+        }
+
+    }
+
+    public async archive(academicYearId) {
+
+        this.moveClass(academicYearId)
```

Comment:
why not await everything so it goes in a sequence as intended instead of jumping ?